doc/nrf/glossary.rst

```diff
@@ -0,0 +1,426 @@
+   |NCS| uses a custom :ref:`Nordic Edition of SES <installing_ses>`.
+
+Signal-to-Noise Ratio (SNR)
+   The level of signal power compared to the level of noise power, often expressed in decibels (dB).
+
+Simple Network Time Protocol (SNTP)
+   A less complex implementation of NTP, using the same protocol but without requiring the storage of state over extended periods of time.
+
+Soft fork
+   A fork that contains a very small set of changes when compared to its upstream.
+
+SoftDevice
+   A wireless protocol stack that complements the nRF5 Series SoCs.
+   Nordic Semiconductor provides these stacks as qualified, precompiled binary files.
+
+SoftDevice Manager (SDM)
+   A SoftDevice component that controls the SoftDevice state and configures the behavior of certain core functionality.
+
+Software Development Kit (SDK)
+   A set of tools used for developing applications for a specific device or operating system.
+
+Spinel
+   A general management protocol for enabling a host device to communicate with and manage co-processors, like a network co-processor (NCP) or a radio co-processor (RCP).
+
+System in Package (SiP)
+   Several integrated circuits, often from different technologies, enclosed in a single module that performs as a system or subsystem.
+
+System on Chip (SoC)
+   A microchip that integrates all the necessary electronic circuits and components of a computer or other electronic systems on a single integrated circuit.
+
+System Protection Unit (SPU)
+   The central point in the system that controls access to memories, peripherals, and other resources.
+
+Target
+   The goal of an operation, for example, programming a specific image on a device, compiling a specific set of files, or removing previously generated files.
+
+Transmission Control Protocol (TCP)
+   A connection-oriented protocol that provides reliable transport.
+   This reliability comes at the cost of control packets overhead of the protocol itself, making it unsuitable for bandwidth-constrained applications.
+
+Toolchain
+   A set of development tools.
+   See `GNU Arm Embedded Toolchain`_.
+
+Transmit Data (TXD)
+   A signal line in a serial interface that transmits data to another device.
+
+Unicast addressing
+   An addressing type that uses a one-to-one association between the destination address and the network endpoint.
+   Each destination address uniquely identifies a single receiver endpoint.
+
+Universal Asynchronous Receiver/Transmitter (UART)
+   A hardware device for asynchronous serial communication between devices.
+
+Universal Serial Bus (USB)
+   An industry standard that establishes specifications for cables and connectors and protocols for connection, communication, and power supply between computers, peripheral devices, and other computers.
+
+Upmerge
+   The act of updating a downstream repository with a new revision of its upstream counterpart.
+
+Upstream
+   The repository from which a downstream is forked off.
```

Comment:
```
Downstream
   The repository that is forked off an upstream.

[...]

Upstream
   The repository from which a downstream is forked off.
```
I see what you did there ;) but one of them should have a precise definition. I'd suggest the upstream

Reply:
Actually, these definitions were already present in the original version of the glossary, used in the application development section. Btw, you are right, I'll clarify them.

Reply:
@jciupis Please check if the new definitions (split in branches and repositories) are more complete.

Reply:
I've checked them, they look good to me now :)